mempool/src/shared_mempool/types.rs

```diff
@@ -250,13 +252,77 @@ impl Ord for BatchId {
     }
 }
 
+#[derive(Clone, Debug, Deserialize, Eq, Hash, PartialEq, Serialize)]
+pub struct TimelineId(pub Vec<u64>);
```

Comment:
we probably should call it MultiBucketTimelineIndexId to avoid confusion and name the field instead of using .0